realsense/scene_perception/win/scene_perception_object.cc

```diff
@@ -1306,23 +1258,30 @@ void ScenePerceptionObject::OnGetSample(
 
   sensemanager_thread_.message_loop()->PostTask(
       FROM_HERE,
-      base::Bind(&ScenePerceptionObject::DoCopySample,
+      base::Bind(&ScenePerceptionObject::DoGetSample,
                  base::Unretained(this),
                  base::Passed(&info)));
 }
 
-void ScenePerceptionObject::DoCopySample(
+void ScenePerceptionObject::DoGetSample(
     scoped_ptr<XWalkExtensionFunctionInfo> info) {
   DCHECK_EQ(sensemanager_thread_.message_loop(), base::MessageLoop::current());
-  if (!(latest_color_image_ && latest_depth_image_)) {
-    Sample sample;
-    info->PostResult(GetSample::Results::Create(
-        sample, std::string("No valiable sample.")));
+
+  pxcStatus status = sense_manager_->AcquireFrame(true);
```

Comment:
`AcquireFrame()` will feed the next frame. I guess `ReleaseFrame()` release the sample internally. If this is the case, we might need to fallback to save sample images in `RunPipeline`.